test/e2e/basic_provisioning_test.go

```diff
@@ -80,7 +69,7 @@ var _ = Describe("BMH Provisioning and Annotation Management", func() {
 				BootMACAddress: bootMacAddress,
 			},
 		}
-		err = clusterProxy.GetClient().Create(ctx, &bmh)
+		err := clusterProxy.GetClient().Create(ctx, &bmh)
 		Expect(err).NotTo(HaveOccurred())
 
 		By("Waiting for the BMH to become available")
```

Comment:
UndesiredStates skipped down here on purpose

Reply:
Excuse me, but I don't get what you mean?

Reply:
I guess he means that we do not use `UndesiredStates` here. It should be fine. We do not need to check all the time that the BMHs don't do weird things like jump from registering to deprovisioning and then available :joy: 
It is very good that we can check in some cases, like when we are supposed to skip deprovisioning, but here I don't see much need for it.

Reply:
Yeah that what I thought that maybe the function fail to cache registering then if it is some other state then we fail but agree it is not that important as it will fail at the end